### PR TITLE
[CORE-1135] Initial commit.

### PR DESCRIPTION
**Pull request by WesleyPumpkinhead**
This is a work in progress. Please do not merge.

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -193,64 +247,141 @@
...
 
+        _getPublicBookingOptions: function () {
+            var _this = this;
+            var options = [];
+
+            if (this.options['options'] && this.options['options'].length > 0) {
+
+                options = this.options['options'].split(',');
+
+                publicBookingOptions.map(function(option){
```
Comment (damianpudlo) [inline app/index.js]: it's misuse of map function, could be replaced with simple for  loop

```
for(var optionKey in publicBookingOptions){
      var option  = publicBookingOptions[optionKey];

      ....
      ....                 
}
```

File: app/index.js [hunk 2]
```diff
@@ -193,64 +247,141 @@
             }
         },
 
+        _getPublicBookingOptions: function () {
```
Comment (damianpudlo) [inline app/index.js]: function name is misleading - seems to be returning all available options but it returns selected one options

could we change it to _getSelectedPublicBookingOptions

File: app/index.js [hunk 3]
```diff
@@ -193,64 +247,141 @@
...
+
+                publicBookingOptions.map(function(option){
+
+                    var index = options.findIndex(function(opt){
+                        return opt === option.abbr;
+                    });
+
+                    if(index !== -1) {
+                        options[index] = option.name;
+                    }
```
Comment (damianpudlo) [inline app/index.js:291]: we should terminate generation process if user tried to use non existing public booking option e.g.

```
if(index === -1){
   _this.log(errorLogFormat('possible possible public booking options'), optionsAbbreviations);
   process.exit(1);
}

options[index] = option
```

File: app/index.js [hunk 4]
```diff
@@ -260,13 +391,25 @@
...
+                });
+            }
             var done = this.async();
             this.prompt(prompts, function (response) {
                 if (response.companyId) this.companyId = response.companyId;
                 if (response.apiUrl) this.apiUrl = response.apiUrl;
                 if (response.developmentApiUrl) this.developmentApiUrl = response.developmentApiUrl;
                 if (response.stagingApiUrl) this.stagingApiUrl = response.stagingApiUrl;
                 if (response.productionApiUrl) this.productionApiUrl = response.productionApiUrl;
+                if (response.googleMapsKey) this.googleMapsKey = response.googleMapsKey === "optional"? this._getOptionDefaults()['google-maps-key'] : response.googleMapsKey
```
Comment (damianpudlo) [inline app/index.js]: missing comma

File: app/index.js [hunk 5]
```diff
@@ -398,6 +544,9 @@
         },
 
         copySrc: function () {
+
+            this.log(typeof path.join, this.sourceRoot(),  this.type);
```
Comment (damianpudlo) [inline app/index.js]: please remove log

File: app/index.js (hunk 1 above)
Comment (WesleyPumpkinhead) [reply app/index.js]: you added that in, not me =D

**Review by edclements — CHANGES_REQUESTED** [review]
Can this be documented in the readme?

**Review by damianpudlo — APPROVED** [review]

**Pull request merged**